### PR TITLE
StarCoder2

### PR DESCRIPTION
**Pull request by lvwerra**

### CODE REVIEW

File: starcoder2.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+- user: nouamanetazi
+---
+
+# StarCoder2 and The Stack v2
+
+<div class="flex items-center justify-center">
+<img src="https://huggingface.co/datasets/bigcode/admin/resolve/main/sc2-banner.png" alt="StarCoder2">
+</div>
+
+BigCode is releasing StarCoder2 the next generation of transparently trained open code LLMs. All StarCoder2 variants were trained on [The Stack v2](https://huggingface.co/datasets/bigcode/the-stack-v2/), a new larg and high quality code dataset. We release all models, datasets, and the processing as well as training code. Check out the [paper](https://drive.google.com/file/d/17iGn3c-sYNiLyRSY-A85QOzgzGnGiVI3/view?usp=sharing) for details.
```
Comment (loubnabnl) [inline starcoder2.md:17]:
```suggestion
BigCode is releasing StarCoder2 the next generation of transparently trained open code LLMs. All StarCoder2 variants were trained on [The Stack v2](https://huggingface.co/datasets/bigcode/the-stack-v2/), a new large and high quality code dataset. We release all models, datasets, and the processing as well as the training code. Check out the [paper](https://drive.google.com/file/d/17iGn3c-sYNiLyRSY-A85QOzgzGnGiVI3/view?usp=sharing) for details.
```

File: starcoder2.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+StarCoder2 offers three model sizes: a 3 billion-parameter model trained by ServiceNow, a 7 billion-parameter model trained by Hugging Face, and a 15 billion-parameter model trained by NVIDIA with NVIDIA NeMo and trained on NVIDIA accelerated infrastructure:
+
+- [StarCoder2-3B](https://huggingface.co/bigcode/starcoder2-3b) was trained on 17 programming languages from The Stack v2 on 3+ trillion tokens.
+- [StarCoder2-7B](https://huggingface.co/bigcode/starcoder2-7b) was trained on 17 programming languages from The Stack v2 on 3.5+ trillion tokens.
+- [StarCoder2-15B](https://huggingface.co/bigcode/starcoder2-15b) was trained on 600+ programming languages from The Stack v2 on 4+ trillion tokens.
+
+StarCoder2-15B is the best in it's size class and matching 33B+ models in many instances and StarCoder2-3B matches the performance of StarCoder1-15B:
```
Comment (loubnabnl) [inline starcoder2.md:29]:
```suggestion
StarCoder2-15B is the best in its size class and matches 33B+ models on many evaluations. StarCoder2-3B matches the performance of StarCoder1-15B:
```

File: starcoder2.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+<img src="https://huggingface.co/datasets/bigcode/admin/resolve/main/sc2-evals.png" alt="StarCoder2 Evaluation">
+</div>
+
+## What is The Stack v2?
+
+<div class="flex items-center justify-center">
+<img src="https://huggingface.co/datasets/bigcode/admin/resolve/main/stackv2-banner.png" alt="The Stack v2">
+</div>
+
+The Stack v2 is the largest open code dataset suitable for LLM pretraining. The Stack v2 is larger, follows an improved language and license detection procedure, and better filtering heuristics. In addition, the training dataset is grouped by repositories, allowing to train models with repository context.
```
Comment (loubnabnl) [inline starcoder2.md:41]:
```suggestion
The Stack v2 is the largest open code dataset suitable for LLM pretraining. The Stack v2 is larger than The Stack v1, follows an improved language and license detection procedure, and better filtering heuristics. In addition, the training dataset is grouped by repositories, allowing to train models with repository context. 
```

**Review by loubnabnl — COMMENTED** [review]
Looks good! I fixed some typos

File: starcoder2.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Links
+
+### Models
+- [Paper](https://drive.google.com/file/d/17iGn3c-sYNiLyRSY-A85QOzgzGnGiVI3/view?usp=sharing): A technical report about StarCoder2 and The Stack v2.
+- [GitHub](https://github.com/bigcode-project/starcoder2/): All you need to know about using or fine-tuning StarCoder2.
+- [StarCoder2-3B](https://huggingface.co/bigcode/starcoder2-3b): Small StarCoder2 model.
+- [StarCoder2-7B](https://huggingface.co/bigcode/starcoder2-7b): Medium StarCoder2 model.
+- [StarCoder2-15B](https://huggingface.co/bigcode/starcoder2-15b): Large StarCoder2 model.
+
```
Comment (loubnabnl) [inline starcoder2.md:65]: let's add the evaluation leaderboard? https://huggingface.co/spaces/bigcode/bigcode-models-leaderboard

File: starcoder2.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+### Data & Governance
+- [StarCoder2 License Agreement](https://huggingface.co/spaces/bigcode/bigcode-model-license-agreement): The model is licensed under the BigCode OpenRAIL-M v1 license agreement.
+- [StarCoder2 Search](https://huggingface.co/spaces/bigcode/search-v2): Full-text search for code in the pretraining dataset.
+- [StarCoder2 Membership Test](https://stack.dataportraits.org): Blazing fast test if code was present in pretraining dataset.
+
+### Others
+- [VSCode Extension](https://marketplace.visualstudio.com/items?itemName=HuggingFace.huggingface-vscode): Code with StarCoder!
+
```
Comment (philschmid) [inline starcoder2.md:73]:
```suggestion
- [Big Code Models Leaderboard](https://huggingface.co/spaces/bigcode/bigcode-models-leaderboard)
```

**Pull request merged**